### PR TITLE
(PDB-2353) support upgrades from 2.3.x

### PR DESCRIPTION
**Pull request by wkalt**
This reinstates our old migrations and inserts some handling in acceptance
tests for upgrades from 2.3.x in addition to stable.

### CODE REVIEW

**Comment by wkalt** [comment]
relies on https://github.com/puppetlabs/ci-job-configs/pull/919

**Comment by ajroetker** [comment]
Do we want to revert the whole migrations change or do we just want to support migrations from 2.3.7 to 4.0.0?

File: acceptance/helper.rb [hunk 1]
```diff
@@ -248,20 +249,24 @@ def display_last_logs(host)
...
   rescue RuntimeError => e
     display_last_logs(host)
     raise
   end
 
   def get_package_version(host, version = nil)
-    return version unless version.nil?
 
-    version = PuppetDBExtensions.config[:package_build_version].to_s
+    if version == 'latest'
```
Comment (ajroetker) [inline acceptance/helper.rb]: Doesn't look like this clause is needed.

File: acceptance/helper.rb [hunk 2]
```diff
@@ -360,6 +365,7 @@ class { 'puppetdb::master::config':
   end
 
   def print_ini_files(host, confdir='/etc/puppetlabs/puppetdb/conf.d')
```
Comment (ajroetker) [inline acceptance/helper.rb]: should we remove the confdir param here now?

File: acceptance/setup/pre_suite/70_install_released_puppetdb.rb [hunk 3]
```diff
@@ -1,10 +1,20 @@
 # We skip this step entirely unless we are running in :upgrade mode.
-if (test_config[:install_mode] == :upgrade and not test_config[:skip_presuite_provisioning])
+OLDEST_SUPPORTED_UPGRADE="2.3.8"
+
+if ([:upgrade_oldest, :upgrade_latest].include? test_config[:install_mode] and not test_config[:skip_presuite_provisioning])
+  if test_config[:install_mode] == :upgrade_latest
+    install_target = 'latest'
+  else
+    install_target = OLDEST_SUPPORTED_UPGRADE
```
Comment (ajroetker) [inline acceptance/setup/pre_suite/70_install_released_puppetdb.rb]: `install_target = test_config[:install_mode] == :upgrade_latest ? 'latest' : OLDEST_SUPPORTED_UPGRADE`

**Comment by ajroetker** [comment]
looks good to me codewise mod some minor comments and I can upgrade from 2.3.x

**Pull request closed without merging**